### PR TITLE
Add Teleport Cloud instructions to two guides

### PR DESCRIPTION
**Pull request by ptgott**
See #10634 

Restricted Sessions for SSH

  While this already includes a compatibility note, I wanted to ensure
  that users would only see information relevant to the scope they had
  selected. Changes made:

  - Added a tabbed Prerequisites section

  - Removed instructions to install the Auth/Proxy that implicitly
    applied to all scopes

  - Misc style/grammar/clarity edits

BPF Session Recording

- Add scoped tabs
- Merge steps 2 and 3, which both involved examining the audit log to
  verify that Enhanced Session Recording is enabled
- Misc style/grammar/clarity tweaks

### CODE REVIEW

**Review by ulysseskan — APPROVED** [review]
These look good to me.

This wasn't part of the changes, but I wonder if this _should_ be changed in any example teleport.yaml config files:

```
  # Replace with IP of Teleport Auth server.
  - 127.0.0.1:3025
```

Can't the `auth_servers` value also be a fully qualified domain name?  So they wouldn't necessarily need to use the IP.  Maybe say `Replace with IP or fully qualified domain name of the Teleport Auth server`.

**Comment by ptgott** [comment]
@ulysseskan I've suggested using the more vague "address"

File: docs/pages/server-access/guides/bpf-session-recording.mdx [hunk 1]
```diff
@@ -55,19 +121,48 @@ $ uname -r
...
+Set up your Teleport Node with the following content in `etc/teleport.yaml`. 
 
 ```yaml
 # Example config to be saved as etc/teleport.yaml
 teleport:
   nodename: graviton-node
   auth_token: exampletoken
   auth_servers:
-  # Replace with IP of Teleport Auth server.
+  # Replace with the address of the Teleport Auth Server.
```
[the remark this answers is not in the document]
Comment (ptgott) [reply docs/pages/server-access/guides/bpf-session-recording.mdx:165]: I think the advantage of using this approach is that the user can paste everything in this snippet into the config file on the Node, make minor adjustments, and proceed to the next step, without having to navigate through the docs. The current Step 2 is incomplete, so I've fleshed it out.

File: docs/pages/server-access/guides/bpf-session-recording.mdx [hunk 2]
```diff
@@ -117,9 +219,8 @@ ssh_service:
 }
 ```
 
-## Step 3/3. Inspect logs
-
-The resulting enhanced session recording will be shown in [Teleport's Audit Log](../../architecture/authentication.mdx#audit-log).
+If you would like to examine your audit log on the Teleport Auth Service host,
+you can examine the contents of `/var/lib/teleport/log` as shown below:
```
[the remark this answers is not in the document]
Comment (ptgott) [reply docs/pages/server-access/guides/bpf-session-recording.mdx]: Added a Details section about this

File: docs/pages/server-access/guides/bpf-session-recording.mdx [hunk 3]
```diff
@@ -57,19 +48,79 @@ $ uname -r
 </tbody>
 </table>
 
-## Step 1/3. Install and configure Teleport node
+(!docs/pages/includes/tctl.mdx!)
+
+## Background
```
Comment (zmb3) [inline docs/pages/server-access/guides/bpf-session-recording.mdx:53]: It feels like this section is too far down on the page. Most readers aren't going to know by default what "Enhanced Session Recording" is or why they would want it.

Should we move some or all of this to the intro before the pre-requisites so we don't lose readers?

File: docs/pages/server-access/guides/bpf-session-recording.mdx [hunk 4]
```diff
@@ -57,19 +48,79 @@ $ uname -r
...
+
+Teleport Nodes submit SSH session traffic to the Auth Service for
+storage, copying what is echoed in a terminal. These recorded sessions can be
+replayed later via the `tsh play` command or in the Teleport Web UI.
+
+The advantage of this approach is that, since no input is captured, these
+session recordings do not contain passwords that were entered into a terminal.
+
+The disadvantage is that session recordings can be bypassed using several
+techniques:
```
Comment (zmb3) [inline docs/pages/server-access/guides/bpf-session-recording.mdx:63]: Maybe reword this? The sessions are still recorded, so I'm not sure I would describe it as session recording being bypassed.

The disadvantage is really that the users true actions can be hidden or obfuscated in such a way that it's hard to determine what actually happened by watching the recording (even though the recording does accurately reflect the session).

File: docs/pages/server-access/guides/bpf-session-recording.mdx [hunk 5]
```diff
@@ -57,19 +48,79 @@ $ uname -r
...
+  without being captured.
+
+Furthermore, due to their unstructured nature, session recordings are difficult to
+ingest and perform monitoring and alerting on.
+
+Teleport Enhanced Session Recording mitigates all three concerns by providing
+advanced security, greater logging capabilities, and better correlates a user
+with their activities.
+
+## Step 1/2. Configure a Teleport Node
```
Comment (zmb3) [inline docs/pages/server-access/guides/bpf-session-recording.mdx:82]: This whole section is mostly prerequisites and not related to enhanced recording. Is the intent that every guide involving SSH nodes should include steps for installing teleport, creating a join token, and joining the cluster? Or can we just document that process in one place and have guides assume you have a node running?

**Review by zmb3 — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

File: docs/pages/server-access/guides/bpf-session-recording.mdx (hunk 5 above)
Comment (ptgott) [reply docs/pages/server-access/guides/bpf-session-recording.mdx:82]: I wanted to make this more like our Database Access guides (e.g., https://goteleport.com/docs/database-access/guides/mongodb-atlas/) where we include as much information as possible in a single guide so users don't need to break focus to follow the guide (https://goteleport.com/docs/contributing/documentation/style-guide/#how-to-guides).
